test/internet/test-dns.js

```diff
@@ -471,6 +471,67 @@ TEST(function test_lookup_localhost_ipv4(done) {
 });
 
 
+TEST(function test_lookup_all_ipv4(done) {
+  var req = dns.lookup('www.google.com', {all: true, family: 4}, function(err, ips) {
+    if (err) throw err;
+    assert.ok(ips.length > 0);
+    assert.ok(Array.isArray(ips));
+
+    ips.forEach(function(ip) {
+      assert.ok(isIPv4(ip.address));
+      assert.strictEqual(ip.family, 4);
+    });
+
+    done();
+  });
+
+  checkWrap(req);
+});
+
+
+TEST(function test_lookup_all_ipv6(done) {
+  var req = dns.lookup('www.google.com', {all: true, family: 6}, function(err, ips) {
+    if (err) throw err;
+    assert.ok(ips.length > 0);
+    assert.ok(Array.isArray(ips));
+
+    ips.forEach(function(ip) {
+      assert.ok(isIPv6(ip.address));
+      assert.strictEqual(ip.family, 6);
+    });
+
+    done();
+  });
+
+  checkWrap(req);
+});
+
+TEST(function test_lookup_all_mixed(done) {
+  var req = dns.lookup('www.google.com', {all: true}, function(err, ips) {
+    var results4 = 0, results6 = 0;
+    if (err) throw err;
+    assert.ok(ips.length > 0);
+    assert.ok(Array.isArray(ips));
+
+    ips.forEach(function(ip) {
```

Comment:
You can make this check into the following, and then get rid of `results4`, `results6`, and the last 4 assertions.

```javascript
if (isIPv4(ip.address))
  assert.equal(ip.family, 4);
else if (isIPv6(ip.address))
  assert.equal(ip.family, 6);
else
  assert(false);
```